core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -2564,6 +2617,67 @@ private static BsonDocument getDocumentIdFilter(final BsonValue documentId) {
     return new BsonDocument("_id", documentId);
   }
 
+  /**
+   * Returns a clone of the given document, but without forbidden fields (currently just the
+   * document version field).
+   *
+   * @param document The document from which to create a clone without forbidden fields.
+   * @return a clone of the given document without forbidden fields
+   */
+  static BsonDocument withoutForbiddenFields(final BsonDocument document) {
+    if (document == null) {
+      return null;
+    }
+
+    final BsonDocument filteredDoc = document.clone();
+    filteredDoc.remove(DOCUMENT_VERSION_FIELD);
```

Comment:
should this be driven by a Collection of constants so that we can easily add more things here should we need to without adding individual calls to "remove"?